components/template-manager/org.wso2.carbon.event.template.manager.core/src/main/java/org/wso2/carbon/event/template/manager/core/internal/util/TemplateManagerHelper.java

```diff
@@ -238,24 +252,121 @@ public static void deployArtifacts(ScenarioConfiguration configuration,
         }
     }
 
+    /**
+     * Create a JavaScript engine packed with given scripts. If two scripts have methods with same name,
+     * later method will override the previous method.
+     *
+     * @param domain
+     * @return JavaScript engine
+     * @throws TemplateDeploymentException if there are any errors in JavaScript evaluation
+     */
+    public static ScriptEngine createJavaScriptEngine(Domain domain) throws TemplateDeploymentException {
+
+        ScriptEngineManager scriptEngineManager = new ScriptEngineManager();
+        ScriptEngine scriptEngine = scriptEngineManager.getEngineByName(TemplateManagerConstants.JAVASCRIPT_ENGINE);
```

Comment:
@lgobinath It seems we have hard coded the script engine name as a constant and set it to Nashorn. Will this work if we run with Java7? Also I think the method can return null as well for the script engine. Therefore a null check for the scriptEngine might make sense before it is used. WDYT?